src/dos/drive_local.cpp

```diff
@@ -616,12 +613,14 @@ bool localFile::Seek(uint32_t *pos, uint32_t type)
 	//TODO Give some doserrorcode;
 		return false;//ERROR
 	}
-	if (!fseek_to_and_check(static_cast<long>(*pos), seektype)) {
+
+	if (!fseek_to_and_check(*reinterpret_cast<int32_t *>(pos), seektype)) {
```

Comment:
I don't like this reinterpret cast in here; on Linux `long` can store all values passed via `uint32_t`, so we don't have a problem there.

The problem exists only for Windows, as `long` == `int` in there…

So the solution in here could be to introduce "yet another wrapper" `cross_fseek` and `cross_ftell`, that would work as `_fseeki64` `_ftelli64` on Windows (but please don't, just listing it as an option for future). However, I consider the real problem to be a bad interface of `localFile::Seek`…

We *can't* fix this interface without touching large number of lines of code, therefore I recommend:

```suggestion
	assert(int64_t(*pos) <= LONG_MAX);
	const auto fpos = static_cast<long>(*pos);
	if (!fseek_to_and_check(fpos, seektype)) {
```
This means, that we effectively don't support files larger than 2GiB - which is fine for now, this is max file size for FAT16 anyway.

Fixing this mess is probably something we could do in the future, but it's low priority, high regression risk, and not part of this PR for sure (and definitely not for 0.76.0) - therefore I recommend adding an assert.

Reply:
I wasn't aware of this 'long' size difference between OSes -- thanks for catching this @dreamer and I agree w/ your reasoning.

Reply:
This re-introduces the regression in Azrael:

``` c++
	assert(int64_t(*pos) <= LONG_MAX);
	const auto fpos = static_cast<long>(*pos);
	if (!fseek_to_and_check(fpos, seektype)) 
```

Reply:
Ok.. all in order now - I think it's documented well enough, and we're doing the right thing with regard to this "int" position.